piccolo/apps/migrations/auto/migration_manager.py

```diff
@@ -739,16 +770,17 @@ async def run_backwards(self):
 
         async with engine.transaction():
 
-            for raw in self.raw_backwards:
-                if inspect.iscoroutinefunction(raw):
-                    await raw()
-                else:
-                    raw()
-
-            await self._run_add_columns(backwards=True)
-            await self._run_add_tables(backwards=True)
-            await self._run_drop_tables(backwards=True)
-            await self._run_rename_tables(backwards=True)
-            await self._run_drop_columns(backwards=True)
-            await self._run_rename_columns(backwards=True)
-            await self._run_alter_columns(backwards=True)
```

Comment:
It looks good - my only concern was this bit.

You've merged `run` and `run_backwards` - I can understand why, as it reduces repetition. The order of the operations were slightly different before when running backwards. Having thought about it though, I think the new order should also be fine.